src/core/qgsexpression.cpp

```diff
@@ -5026,6 +5026,17 @@ QString QgsExpression::formatPreviewString( const QVariant& value )
       return previewString.prepend( '\'' ).append( '\'' );
     }
   }
+  else if ( value.type() == QVariant::Map )
+  {
+    QString mapStr;
+    const QVariantMap map = value.toMap();
+    for ( QVariantMap::const_iterator it = map.constBegin(); it != map.constEnd(); ++it )
```

Comment:
I'd suggest only showing the first 2 or 3 values and then truncating the rest with an "...". Part of the function of formatPreviewString() is to avoid returning excessively long strings (which were mangling the builder widget and dialog)